src/torchmetrics/classification/jaccard.py

```diff
@@ -15,9 +15,246 @@
+        average:
+            Defines the reduction that is applied over labels. Should be one of the following:
+
+            - ``micro``: Sum statistics over all labels
+            - ``macro``: Calculate statistics for each label and average them
+            - ``weighted``: Calculates statistics for each label and computes weighted average using their support
+            - ``"none"`` or ``None``: Calculates statistic for each label and applies no reduction
+
+        validate_args: bool indicating if input arguments and tensors should be validated for correctness.
+            Set to ``False`` for faster computations.
+        kwargs: Additional keyword arguments, see :ref:`Metric kwargs` for more info.
+
+    Example (pred is integer tensor):
+        >>> from torchmetrics import MulticlassJaccardIndex
+        >>> target = torch.tensor([2, 1, 0, 0])
+        >>> preds = torch.tensor([2, 1, 0, 1])
+        >>> metric = MulticlassJaccardIndex(num_classes=3)
+        >>> metric(preds, target)
+        tensor(0.6667)
+
+    Example (pred is float tensor):
+        >>> from torchmetrics import MulticlassJaccardIndex
+        >>> target = torch.tensor([2, 1, 0, 0])
+        >>> preds = torch.tensor([
+        ...   [0.16, 0.26, 0.58],
+        ...   [0.22, 0.61, 0.17],
+        ...   [0.71, 0.09, 0.20],
+        ...   [0.05, 0.82, 0.13],
+        ... ])
+        >>> metric = MulticlassJaccardIndex(num_classes=3)
+        >>> metric(preds, target)
+        tensor(0.6667)
+    """
+
+    is_differentiable: bool = False
+    higher_is_better: bool = True
+    full_state_update: bool = False
+
+    def __init__(
+        self,
+        num_classes: int,
+        ignore_index: Optional[int] = None,
+        average: Optional[Literal["micro", "macro", "weighted", "none"]] = "macro",
+        validate_args: bool = True,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(
+            num_classes=num_classes, ignore_index=ignore_index, normalize=None, validate_args=False, **kwargs
+        )
+        if validate_args:
+            _multiclass_jaccard_index_arg_validation(num_classes, ignore_index, average)
+        self.validate_args = validate_args
+        self.average = average
+
+    def compute(self) -> Tensor:
+        return _jaccard_index_reduce(self.confmat, average=self.average)
+
+
+class MultilabelJaccardIndex(MultilabelConfusionMatrix):
+    r"""Calculates the Jaccard index for multilabel tasks. The `Jaccard index`_ (also known as
+    the intersetion over union or jaccard similarity coefficient) is an statistic that can be
+    used to determine the similarity and diversity of a sample set. It is defined as the size
+    of the intersection divided by the union of the sample sets:
+
+    .. math:: J(A,B) = \frac{|A\cap B|}{|A\cup B|}
+
+    Accepts the following input tensors:
+
+    - ``preds`` (int or float tensor): ``(N, C, ...)``. If preds is a floating point tensor with values outside
```

Comment:
```suggestion
    - ``preds``: ``(N, C, ...)``. If preds is a floating point tensor with values outside
```
this shall come from function footprint, right so I may skip it...
in case we have it elsewhere, lets prune/simplify in another PR package-wide